sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -425,8 +425,8 @@ internal virtual async Task<Response> DeleteAsync(string key, string label, Cond
 
                 return response.Status switch
                 {
-                    200 => response,
-                    204 => response,
+                    200 => CreateResponse(response),
+                    204 => CreateResourceModifiedResponse(response),
```

Comment:
If we move forward with this, we'll want to rename CreateResourceModifiedResponse to CreateNoBodyResponse.